### PR TITLE
Show warnings even if no changes

### LINKED ISSUE #22541: Terraform 0.12 shows warnings only when there are changes
### Terraform Version

```
terraform version
Terraform v0.12.6
```

### Terraform Configuration Files

main.tf
```hcl
provider "null" {

}
resource "null_resource" "res" {
}
```
main.tfvars
```hcl
test_var="12345"
```
### Run
```bash
terraform init && terraform plan -var-file=main.tfvars
```

### Expected Behavior

Warnings should be shown even if there are no changes.

### Actual Behavior

Warnings are shown only when there are changes.

### Steps to Reproduce

1. `terraform init`
2. `terraform plan -var-file=main.tfvars`

Changes and warnings are shown

1. `terraform apply -var-file=main.tfvars -auto-approve`
2. `terraform plan -var-file=main.tfvars`

No changes, no warnings

### Additional Context

After migration to terraform 0.12 we saw stricter behavior and warnings we would like to address, but we see them only when there are changes. So we have to create a change artificially in order to fix those recommendations. 

Example of the warning:

```
Warning: Value for undeclared variable

  on main.tfvars line 1:
   1: test_var="12345"

The root module does not declare a variable named "test_var". To use this
value, add a "variable" block to the configuration.

Using a variables file to set an undeclared variable is deprecated and will
become an error in a future release. If you wish to provide certain "global"
settings to all configurations in your organization, use TF_VAR_...
environment variables to set these instead.
```

### PR DESCRIPTION
**Pull request by nyurik**
Fixes #22541

This change will print warnings, such as deprecations,
even if plan has no changes.

### CODE REVIEW

File: backend/local/backend_plan.go [hunk 1]
```diff
@@ -162,19 +162,20 @@ func (b *Local) opPlan(
 
 		if plan.Changes.Empty() {
 			b.CLI.Output("\n" + b.Colorize().Color(strings.TrimSpace(planNoChanges)))
-			return
+		} else {
```
Comment (apparentlymart) [inline backend/local/backend_plan.go]: This new approach means that everything after this point will need to remember to guard for `plan.Changes.Empty()`, which seems likely to lead to mistakes as this code evolves.

What do you think about instead just adding a new call to `b.ShowDiagnostics` just before the `return`, so it'll still happen in both codepaths but the rest of this function can assume there are changes to make?

Comment (nyurik) [reply backend/local/backend_plan.go]: @apparentlymart sure, thx, updated

**Comment by nyurik** [comment]
@apparentlymart thanks, I updated the PR. Is there anything else that needs to be done here, or any specific person to ping to review it? Thanks!

**Comment by apparentlymart** [comment]
Thanks for the updates, @nyurik! This looks good to me, and I'm going to merge it.

**Comment by nyurik** [comment]
@apparentlymart would it be possible to backport this to 0.11 branch?  This way it will make migration 0.11 -> 0.12 much easier -- we would simply need to migrate to the latest of the 0.11 branch, and our terraforms will instantly start showing any kind of upgrade warnings - thus we will be able to fix them before actually trying to migrate.

**Comment by nyurik** [comment]
Update -- seems that 0.11 had a different output code. I did make a backport for v0.12 #24169 - hopefully can be merged & published soon.  Thanks!

**Pull request merged**